backend/src/main/java/com/celuveat/celuveat/restaurantlinkfetcher/infra/youtube/YouTubeRestaurantLinkFetcher.java

```diff
@@ -0,0 +1,88 @@
+package com.celuveat.celuveat.restaurantlinkfetcher.infra.youtube;
+
+import com.celuveat.celuveat.restaurantlinkfetcher.domain.RestaurantLinkFetcher;
+import com.celuveat.celuveat.restaurantlinkfetcher.infra.youtube.api.YouTubeDataApi;
+import com.celuveat.celuveat.restaurantlinkfetcher.infra.youtube.dto.search.SearchListResponse;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class YouTubeRestaurantLinkFetcher implements RestaurantLinkFetcher {
+
+    private final YouTubeDataApi youTubeDataApi;
+
+    @Override
+    public List<String> fetchAllByChannelId(String channelId) {
+        SearchListResponse response = youTubeDataApi.searchList(channelId);
+        List<String> videoIds = new ArrayList<>(getVideoIds(response));
```

Comment:
new ArrayList()는 지워도 될 것 같아
```suggestion
        List<String> videoIds = getVideoIds(response);
```

Reply:
stream에서 toList() 받아오는게 Immutable이라 new ArrayList<>()로 복사 안하면 데이터 추가를 못해서 감쌌어.
지금 API에서 한번에 받아올 수 있는 데이터가 50개가 최대라 셀럽 전체 영상을 받아오려면 API 호출을 여러번 해야해서 `videoIds`에 `addAll`하는 게 불가피해 보이는데 혹시 다르게 해결하는 방법이 있을까?

Reply:
말로 했으니 닫을게!